packages/datagrid/src/components/DataGrid/DataGrid.component.js

```diff
@@ -80,6 +80,30 @@ export function getRowDataInfos(rowData) {
 	);
 }
 
+/**
+ * redrawRows - call redrawRows only if necessary. (improve ag-grid performance)
+ *
+ * @param  {object} props     component props
+ * @param  {object} gridAPI   ag-grid api
+ * @param  {object} prevProps previous component prop
+ */
+export function redrawRows(props, gridAPI, prevProps) {
+	if (props.loading || !gridAPI) {
```

Comment:
you can revert the conditional and put the code in this if statement instead of using an early return

Reply:
What's problematic with early returns ? here or in general.

Reply:
i prefer this rather that a condition pyramid or huge block.